[Docs] Update flashing information to include :flash target (#6999)
qmk/qmk_firmware: docs/newbs_flashing.md

## Flash your Keyboard from the Command Line

This has been made pretty simple compared to what it used to be.  When you are ready to compile and flash your firmware, open up your terminal window and run the build command: 

    make <my_keyboard>:<my_keymap>:flash

For example, if your keymap is named "xyverz" and you're building a keymap for a rev5 planck, you'll use this command:

    make planck/rev5:xyverz:flash

This will check the keyboard's configuration, and then attempt to flash it based on the specified bootloader.  This means that you don't need to know which bootloader that your keyboard uses.  Just run the command, and let the command do the heavy lifting.

However, this does rely on the bootloader being set by the keyboard.  If this information is not configured, or you're using a board that doesn't have a supported target to flash it, you will see this error:

    WARNING: This board's bootloader is not specified or is not supported by the ":flash" target at this time.

In this case, you'll have to fall back on specifying the bootloader.

> There are five main bootloaders that are used, usually. Pro-Micro and clones use Caterina, and Teensy's use Halfkay, OLKB's AVR boards use QMK-DFU, other atmega32u4 chips use DFU, and most ARM boards use ARM DFU.

Reviewer: ```suggestion
There are five main bootloaders that are used. Pro Micro and clones use Caterina, Teensys use HalfKay, OLKB's AVR boards use QMK-DFU, other ATmega32U4 boards use DFU, and most ARM boards use ARM DFU.
```

    sudo make <my_keyboard>:<my_keymap>:avrdude

#### Caterina commands

There are a number of DFU commands that you can use to flash firmware to a DFU device:

> * `:avrdude` - This is the normal option and waits until a caterina device is available, and then flashes the firmware. This will wait until it detects a new COM port, and then it will flash.

Reviewer: ```suggestion
* `:avrdude` - This is the normal option which waits until a Caterina device is available (by detecting a new COM port), and then flashes the firmware.
```

> * `:avrdude-loop` - This runs the same command as `:avrdude`, but after each device is flashed, it will attempt to flash again.  This is useful for bulk flashing. _This requires you to manually escape the loop by hitting Ctrl+C, Cmd+C, or whatever the hotkey is for your system._

Reviewer: ```suggestion
* `:avrdude-loop` - This runs the same command as `:avrdude`, but after each device is flashed, it will attempt to flash again.  This is useful for bulk flashing. _This requires you to manually escape the loop by hitting Control+C._
```
